### ISSUE TITLE
Implement a conservative auto-repair operation in cases where a database file is unable to be opened or read due to corrupted data.

### ISSUE THREAD

**Issue by dev7355608** [foundryvtt/foundryvtt#9824]
## Original Issue Below

Originally reported by code_phi: https://discord.com/channels/170995199584108546/1065764070680186930/1131215544213844069

**ALL MODULES DISABLED?** false
**OS, Hosting, Browser (if applicable)**: Forge hosting
**Short Description of bug**: LevelDB corruption: CURRENT file may not point to latest MANIFEST file
**Simple steps to reproduce the bug**:
- Difficult to reproduce, unfortunately. Related to LevelDB corruption from uncertain sources. 
- Normal functioning of the worlds. After an hour's idle time, the process is stopped, but Return to Setup is never clicked.
- Note: Closing the Foundry process does not call exit handler in the same way that clicking Return to Setup would. This means that Foundry's LevelDatabase wrapper `close` function is not called and database compaction does not happen on close. It's possible that this, or earlier, is where the corruption came in.
- LevelDB CURRENT file, which should point to the MANIFEST file in LevelDB, was not updated. It points to an old MANIFEST file which no longer exists. Note that Forge does not lock the CURRENT file or interfere with the LevelDB database. This was captured on Forge and is rare, but it might also be happening elsewhere.
- This LevelDB database can no longer be opened until the CURRENT file is updated manually to point to the latest MANIFEST files. The error is `Failed to connect to database "scenes": Database not open`. Please see screenshots.
- A possible solution might be call `ClassicLevel.repair` on the LevelDB database location if the database cannot be opened with a similar error or for any unknown reason.
- The `ClassicLevel.repair` process also performs compaction after closing the database. It might be helpful to incorporate repair into the close flow somehow.
- For the case shown in the screenshots, both the scenes and actors DBs were corrupted in the same way. We've also received reports of various modules or systems being unable to open compendiums until a re-install of that module or system, which then fixes it. It may be a similar kind of corruption.
**Screenshots and/or console errors**:
![image](https://github.com/foundryvtt/foundryvtt/assets/31905376/6af44294-60c8-40a0-b389-2b22df3e3b40)
![image](https://github.com/foundryvtt/foundryvtt/assets/31905376/b83635e3-e452-4f51-8373-5c8989933ff6)

**Comment by aaclayton** [comment]
Perhaps related to #9603, also nonreproducible

**Comment by rikmarais** [comment]
Some more information regarding this issue:
We have encountered another case where the above error has happened. In this instance it was due to an Import of that data to The Forge while Foundry was also running on the user's local computer.

Some of the other cases where we encountered did not involve an Import while Foundry was running, but it is understandably difficult to reproduce.

**Comment by rikmarais** [comment]
We recently encountered another case of database corruption where the `actors` db's CURRENT file was pointing to an incorrect MANIFEST.

To assist the user, I asked them if they would be willing to provide me with a copy of the database, and I ran `ClassicLevel`'s `repair` function on it, which recovered the data and made the database accessible.

As mentioned here https://github.com/foundryvtt/foundryvtt/issues/9824#issue-1818124935
> Closing the Foundry process does not call exit handler in the same way that clicking Return to Setup would. This means that Foundry's LevelDatabase wrapper close function is not called and database compaction does not happen on close

> A possible solution might be call ClassicLevel.repair on the LevelDB database location if the database cannot be opened with a similar error or for any unknown reason.
The ClassicLevel.repair process also performs compaction after closing the database. It might be helpful to incorporate repair into the close flow somehow.

There's little recourse for users unfortunate enough to encounter db corruption. I think there may be benefit to including `repair` in the workflow somewhere when encountering a database that cannot be opened, possibly as a prompt (or as [part of the CLI tool](https://github.com/foundryvtt/foundryvtt-cli/issues/32))
